snuba/datasets/entities/transactions.py

```diff
@@ -111,21 +124,61 @@ def select_storage(
         return StorageAndMappers(storage, self.__mappers)
 
 
+def v2_selector_function(query: Query, referrer: str) -> Tuple[str, List[str]]:
+    if settings.TRANSACTIONS_UPGRADE_BEGINING_OF_TIME is None or not isinstance(
+        query, ProcessableQuery
+    ):
+        return ("transactions_v1", [])
+
+    range = get_time_range(query, "timestamp")
+    if range[0] is None or range[0] < settings.TRANSACTIONS_UPGRADE_BEGINING_OF_TIME:
+        return ("transactions_v1", [])
+
+    mapping = {
+        Option.TRANSACTIONS: "transactions_v1",
+        Option.TRANSACTIONS_V2: "transactions_v2",
+    }
+    choice = RolloutSelector(
+        Option.TRANSACTIONS, Option.TRANSACTIONS_V2, "transactions"
+    ).choose(referrer)
+    if choice.secondary is None:
+        return (mapping[choice.primary], [])
+    else:
+        return (mapping[choice.primary], [mapping[choice.secondary]])
+
+
 class BaseTransactionsEntity(Entity, ABC):
     def __init__(self, custom_mappers: Optional[TranslationMappers] = None) -> None:
         storage = get_writable_storage(StorageKey.TRANSACTIONS)
         schema = storage.get_table_writer().get_schema()
+        mappers = (
+            transaction_translator
+            if custom_mappers is None
+            else transaction_translator.concat(custom_mappers)
+        )
 
-        pipeline_builder = SimplePipelineBuilder(
+        v1_pipeline_builder = SimplePipelineBuilder(
             query_plan_builder=SelectedStorageQueryPlanBuilder(
-                selector=TransactionsQueryStorageSelector(
-                    mappers=transaction_translator
-                    if custom_mappers is None
-                    else transaction_translator.concat(custom_mappers)
-                )
+                selector=TransactionsQueryStorageSelector(mappers=mappers)
             ),
         )
 
+        v2_pipeline_builder = SimplePipelineBuilder(
+            query_plan_builder=SingleStorageQueryPlanBuilder(
+                storage=get_writable_storage(StorageKey.TRANSACTIONS_V2),
```

Comment:
You just need a readable storage, `get_writable_storage` return a writable one. You can go with `get_storage`